src/components/structures/LeftPanel.js

```diff
@@ -209,6 +212,14 @@ const LeftPanel = createReactClass({
         this._roomList = ref;
     },
 
+    _onSearchFocus: function() {
+        this.setState({searchFocused: true});
+    },
+
+    _onSearchBlur: function() {
+        this.setState({searchFocused: false});
```

Comment:
should we keep it expanded if there's text in the field?

Reply:
Yeah, good point.